File: lab-3-ddb/environment/python_3.6.8/add_index.py
import boto3

client = boto3.<FMI>("dynamodb", region_name="us-east-1")

res = <FMI>.update_table(
    AttributeDefinitions=[
        {
                "<FMI>": "breed",
                "<FMI>": "S"
        },
    ], 
    TableName="lostcats",
    GlobalSecondaryIndexUpdates=[
        {
            'Create': {
                'IndexName': 'breed_index',
                'KeySchema': [
                    {
                        'AttributeName': 'breed',
                        'KeyType': 'HASH'
                    }
                ],
                'Projection': {
                    'ProjectionType': 'ALL'
                },
                '<FMI>': {
                    'ReadCapacityUnits': 1,
                    'WriteCapacityUnits': 1
                }
            }
        }
    ],
)
print(res)

File: lab-3-ddb/environment/ruby_2.6.0/solution/add_index.rb
require "aws-sdk-dynamodb"
client = Aws::DynamoDB::Client.new(region: "us-east-1")

res = client.update_table(
    attribute_definitions:[
        {
                attribute_name: "breed",
                attribute_type: "S"
        },
    ], 
    table_name: "lostcats",
    global_secondary_index_updates:[
        {
            create: {
                index_name: 'breed_index',
                key_schema: [
                    {
                        attribute_name: 'breed',
                        key_type: 'HASH'
                    }
                ],
                projection: {
                    projection_type: 'ALL'
                },
                provisioned_throughput: {
                    read_capacity_units: 1,
                    write_capacity_units: 1
                }
            }
        }
    ],
)